fuse/man/fuse.1
.\" -*- nroff -*-
.\"
.\" fuse.1: Fuse man page
.\" Copyright (c) 2001-2016 Russell Marks, Philip Kendall, Darren Salt,
.\"                         Fredrick Meunier, Stuart Brady
.\"
.\" This program is free software; you can redistribute it and/or modify
.\" it under the terms of the GNU General Public License as published by
.\" the Free Software Foundation; either version 2 of the License, or
.\" (at your option) any later version.
.\"
.\" This program is distributed in the hope that it will be useful,
.\" but WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License along
.\" with this program; if not, write to the Free Software Foundation, Inc.,
.\" 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
.\"
.\" Author contact information:
.\"
.\" E-mail: philip-fuse@shadowmagic.org.uk
.\"
.\"
.TH fuse 1 "21st August, 2016" "Version 1.2.2" "Emulators"
.\"
.\"------------------------------------------------------------------
.\"
.SH NAME
fuse \(em Sinclair ZX Spectrum emulator
.\"
.\"------------------------------------------------------------------
.\"
.SH SYNOPSIS
.\" the trend for long-option-using programs is to give a largely
.\" generic synopsis, so...
.B fuse
.RI [ options ]
.P
.\"
.\"------------------------------------------------------------------
.\"
.SH DESCRIPTION
Fuse is a Sinclair ZX\ Spectrum emulator. It supports several models
(including the 128), with quite faithful emulation of the display and
sound.
.PP
The emulator can load any of the formats supported by
.IR libspectrum (3)
\(em this includes Z80, SNA and SZX snapshots, and PZX, TAP and TZX
virtual-tape files. Saving to SZX, Z80 and SNA snapshots and TZX and
TAP tape files is supported. The SLT extension to the Z80 format is
partly supported (enough for multi-load games); however, loading of
the old DAT-file variant is not.
.PP
DSK, UDI, FDI, TD0, MGT, IMG, D40, D80, SAD, TRD, SCL and OPD
disk images are supported when a disk interface is being emulated,
including the integrated disk drives on +3, Pentagon or Scorpion
machines as well as the +D, DISCiPLE, Opus Discovery, Beta\ 128 and
Didaktik\ 80 interfaces. DCK cartridge images are supported when
emulating a Timex\ 2068 variant. Interface\ 2 ROM cartridges are
also supported.
.PP
Finally, there is also support for reading and writing the RZX input recording
format.
.PP
See the
.B "COMPRESSED FILES"
section for details on reading files compressed with
.IR bzip2 "(3),"
.IR gzip "(3)"
or
.IR zip "(3)."
.\"
.\"------------------------------------------------------------------
.\"
.SH OPTIONS
.\" dual short/long option listings here reflect the GNU approach,
.\" as used in info files. It does mean using RS/RE, though,
.\" so this is used for all options (for consistency).
.\"
.\" the options list is in alphabetical order by long option name (or
.\" short option name if none).
.\"
.B \-\-accelerate\-loader
.RS
Specify whether Fuse should attempt to accelerate tape loaders by \(lqshort
circuiting\(rq the loading loop. This will in general speed up loading, but
may cause some loaders to fail. (Enabled by default, but you can use
.RB ` \-\-no\-accelerate\-loader '
to disable). The same as the Media Options dialog's
.I "Accelerate loaders"
option.
.RE
.PP
.B \-\-aspect\-hint
.RS
Specify whether the GTK+ and Xlib user interfaces should `hint' to the
window manager about the preferred aspect ratio for the graphics
window, thus preventing resizing to non-square sizes which lead to
Fuse not displaying correctly. This option has been observed to cause
problems with some window managers when using the GTK+ UI which can
prevent the window from being resized or moved at all. (Enabled by
default, but you can use
.RB ` \-\-no\-aspect\-hint '
to disable). See also the
.RB ` \-\-strict\-aspect\-hint '
option.
.RE
.PP
.B \-\-autosave\-settings
.RS
Specify whether Fuse's current settings should be automatically saved
on exit. The same as the General Options dialog's
.I "Auto-save settings"
option.
.RE
.PP
.B \-\-auto\-load
.RS
Specify whether tape and disk files should be automatically loaded
when they are opened using the
.I "File, Open..."
menu option. In the case of TRD/SCL disk images, inserts also a boot
loader file when none is available. (Enabled by default, but you can use
.RB ` \-\-no\-auto\-load '
to disable). Same as the Media Options dialog's
.I "Auto-load media"
option.
.RE
.PP
.B \-\-beta128
.RS
Emulate a Beta\ 128 interface. Same as the Disk Peripherals Options dialog's
.I "Beta\ 128 interface"
option.
.RE
.PP
.B \-\-beta128\-48boot
.RS
When a Beta\ 128 interface is used in 48K or TC2048 emulation the
option additionally controls whether the machine boots directly into the TR-DOS
system. Same as the Disk Peripherals Options dialog's
.I "Beta\ 128 auto-boot in 48K machines"
option.
.RE
.PP
.B \-\-betadisk
.I file
.RS
Insert the specified file into the emulated Beta disk interface's drive\ A:
and select Pentagon mode on startup.
.RE
.PP
.B \-\-bw\-tv
.RS
Specify whether the display should simulate a colour or black and
white television. This option is effective under the GTK+, Win32, Xlib
and SDL user interfaces: the others will always simulate a colour TV.
The same as the General Options dialog's
.I "Black and white TV"
option.
.RE
.PP
.B \-\-cmos-z80
.RS
This option specifies that Fuse should emulate a CMOS Z80, as opposed
to an NMOS Z80. Same as the General Options dialog's
.I "Z80 is CMOS"
option.
.RE
.PP
.B \-\-competition\-code
.I code
.RS
Specify the code to be written to competition mode RZX files. The same
as the RZX Options dialog's
.I "Competition code"
option.
.RE
.PP
.B \-\-competition\-mode
.RS
Specify whether input recordings should be made in `competition mode'.
The same as the RZX Options dialog's
.I "Competition mode"
option.
.RE
.PP
.B \-\-compress\-rzx
.RS
Specify whether RZX files should be written out compressed. (Enabled
by default, but you can use
.RB ` \-\-no\-compress\-rzx '
to disable). Same as the RZX Options dialog's
.I "Compress RZX data"
option.
.RE
.PP
.B \-\-confirm\-actions
.RS
Specify whether `dangerous' actions (those which could cause data
loss, for example resetting the Spectrum) require confirmation before
occurring. (Enabled by default, but you can use
.RB ` \-\-no\-confirm\-actions '
to disable). This option is the same as the General Options dialog's
.I "Confirm actions"
option.
.RE
.PP
.B \-\-debugger\-command
.I string
.RS
Specify a debugger command to be run before emulator startup. This can
be used to set breakpoints or the like. Currently, this is the only
method to input multi-line debugger commands. (See the
.B "MONITOR/DEBUGGER"
section for more information).
.RE
.PP
.B \-\-detect\-loader
.RS
Specify whether Fuse should attempt to detect when the tape is being
accessed and start and stop the virtual tape playing
automatically. (Enabled by default, but you can use
.RB ` \-\-no\-detect\-loader '
to disable). Same as the Media Options dialog's
.I "Detect loaders"
option.
.RE
.PP
.B \-\-disciple
.RS
Emulate a DISCiPLE interface. Same as the Disk Peripherals Options dialog's
.I "DISCiPLE interface"
option.
.RE
.PP
.B \-\-discipledisk
.I file
.RS
Insert the specified file into the emulated DISCiPLE's drive\ 1.
.RE
.PP
.B \-\-didaktik80
.RS
Emulate a Didaktik 80 (or Didaktik 40) disk interface. Same as the
Disk Peripherals Options dialog's
.I "Didaktik 80 interface"
option.
.RE
.PP
.B \-\-didaktik80disk
.I file
.RS
Insert the specified file into the emulated Didaktik\ 80
(or Didaktik\ 40)'s drive\ A.
.RE
.PP
.B \-\-disk\-ask\-merge
.RS
Prompt the user to confirm whether Fuse should try to merge the `B' side of a
disk image from a separate file when opening a new single-sided disk image.
.RE
.PP
.B \-\-disk\-try\-merge
.I mode
.RS
Select whether Fuse should try to merge a separate file for the `B' side of a
disk image separate file when opening a new disk image. Most double sided disk
images are dumped as two single sided disk images e.g.
`Golden Axe \- Side\ A.dsk' and `Golden Axe \- Side\ B.dsk'. So, if we want to
play Golden Axe, first we have to insert the first disk image and when the game
asks to insert side\ B, we have to find and open the second disk image, instead
of just `flip'-ing the disk inside the drive. If enabled, Fuse will try to open
the second image too and create a double sided disk image (merging the two one
sided disk images) and insert this merged virtual disk into the disk drive. The
function detects whether the file is one side of a double-sided image if the
filename matches a pattern like [Ss]ide[ _][abAB12][ _.] in the file name of a
disk that is being opened. If found, Fuse will try to open the other side of
the disk too substituting the appropriate characters in the filename e.g.
1\(->2, a\(->b, A\(->B. If successful then it will merge the two images and
now we have a double sided disk in drive. This means that if we open
`Golden Axe \- Side\ A.dsk', then Fuse will try to open `Golden Axe \- Side
B.dsk' too. Now, we can just `flip' the disk if Golden Axe asks for `Side\ B'.
The available options are
.IR Never ,
.I "With single-sided drives"
and
.IR Always .
.RE
.PP
.B \-\-divide
.RS
Emulate the DivIDE interface. The same as the Disk Peripherals Options
dialog's
.I "DivIDE interface"
option.
.RE
.PP
.B \-\-divide\-masterfile
.I file
.br
.B \-\-divide\-slavefile
.I file
.RS
Specify an IDE image to be loaded into the DivIDE's emulated master
and slave drives respectively.
.RE
.PP
.B \-\-divide\-write\-protect
.RS
Specify that the emulated DivIDE's write protect jumper should be
considered set. The same as the Disk Peripherals Options dialog's
.I "DivIDE write protect"
option.
.RE
.PP
.B \-\-dock
.I file
.RS
Insert the specified file into the emulated Timex\ 2068 variant dock;
also select the TC2068 on startup if available.
.RE
.PP
.B \-D
.I mode
.br
.B \-\-doublescan\-mode
.I mode
.RS
Specify whether to use doublescan modes in the FB UI.
Available values for
.I mode
are 0, 1 and 2. 0 means `never doublescan' (use 640\(mu480 at either 72\ Hz or
60\ Hz), whereas 1 and 2 both mean `try to use doublescan' and will fall back

fuse/ChangeLog
2016-08-21  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * Fuse 1.2.2 released.

        * New features:
          * Support loading first tape, snapshot, dock cartridge or RZX file
            found inside .zip files (Patrik Rak and Sergio Baldoví).
          * Support auto-booting TR-DOS disk images without a boot file (thanks,
            windale, BogDan Vatra and Fredrick Meunier) (Sergio Baldoví).

        * Emulation core improvements:
          * Change microphone state when 0 tstate pulses do not have the no edge
            flag set (Fredrick Meunier).

        * Machine specific improvements:
          * Fix +3 disk autoload (thanks, windale and BogDan Vatra) (Sergio
            Baldoví and Fredrick Meunier).
          * Fix floppy drive selection when resetting a +3 (thanks, windale and
            BogDan Vatra) (Sergio Baldoví).

        * Miscellaneous improvements:
          * WidgetUI: Use description for tape blocks where available (Fredrick
            Meunier).
          * Use pkg-config to detect libpng and libxml2 (Alberto Garcia).
          * Various minor bugfixes.

2016-07-17  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * Fuse 1.2.1 released.

        * Emulation core improvements:
          * Fix bugs when the detect loaders feature is being used (thanks, Lee
            Tonks and windale) (Fredrick Meunier).

        * Debugger improvements:
          * Remove the need for "%" when accessing system variables (Philip
            Kendall).
          * Add Z80 registers as debugger variables (Philip Kendall).
          * Expose last byte written to the ULA, tstates since interrupt,
            primary and secondary memory control ports as debugger system
            variables (Philip Kendall).
          * Make breakpoints on events honour lifetime (Sergio Baldoví).
          * Extend breakpoints on paging events to more peripherals: Beta 128,
            +D, Didaktik 80, DISCiPLE, Opus Discovery and SpeccyBoot (Sergio
            Baldoví).
          * Split +D memory sources into RAM and ROM sections (Sergio Baldoví).
          * Coalesce +D and DISCiPLE RAM pages so they show as 8K pages (Sergio
            Baldoví).

        * Miscellaneous improvements:
          * Update GNOME .desktop file (Alberto Garcia).
          * Add an emulator module startup manager to automatically handle
            dependency issues (Philip Kendall).
          * Fix crash on widget UIs when hitting the close icon on the title bar
            several times (Sergio Baldoví).

2016-06-06  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * Fuse 1.2.0 released.

        * New features:
          * Add Currah µSource emulation (Stuart Brady).
          * Add Didaktik 80/40 emulation (Gergely Szasz).
          * Capture BASIC video functions to SVG file (Stefano Bodrato).
          * Support bash completion (Sergio Baldoví).

        * Emulation core improvements:
          * Allow continuing RZX recordings if there is a final snapshot in the
            RZX (Sergio Baldoví).
          * Fix the prune function on RZX rollback (Sergio Baldoví).
          * Use SZX format for the initial snapshot in RZX files (Sergio
            Baldoví).
          * Fix loading of EDSK files with Sector Offset block (Sergio Baldoví).
          * Migrate disk "index event" handling to the FDD layer and have the
            FDC layer use it for their STATUS registers (Gergely Szasz).
          * Implement WD2797 emulation (Gergely Szasz).
          * Centralise the "Disk icon" update code to the FDD layer (Gergely
            Szasz).
          * Fix disk image corruption after saving UDI files (Sergio Baldoví).
          * Check ready status after loading a disk into floppy disk drive
            (thanks, John Elliott) (Sergio Baldoví).
          * Fix overlapped SEEK commands (Sergio Baldoví).
          * Fix length of data returned by READ_DIAG (thanks, Fredrick Meunier)
            (Sergio Baldoví).
          * Fix writing .td0 format disk files (Sergio Baldoví).
          * Fix speech in Cobra's Arc - Medium Case.tzx when loaded with tape
            traps enabled (thanks, zx81 and Sergio Baldoví) (Fredrick Meunier).
          * Skip tape traps if VERIFY is requested (UB880D).
          * Fix loading sound with some custom loaders (Fredrick Meunier).
          * Check if data blocks are headers for handling PZX files (Fredrick
            Meunier).
          * Correct display of pulses in PZX pulse block for tape browser
            (Fredrick Meunier).
          * Set AF, AF' and SP to 0xffff on reset (Stuart Brady).
          * Leave most registers unchanged on a soft reset (Stuart Brady).
          * Emulate interrupt and NMI timings more precisely (Stuart Brady).
          * Emulate NMOS and CMOS Z80 variants (Stuart Brady).
          * Switch to 2KB page size (Stuart Brady).
          * Fix inaccurate output when AY envelopes are used (Matthew Westcott
            and Fredrick Meunier).
          * Fix some peripherals activation when loading snapshots (Sergio
            Baldoví).

        * Machine specific improvements:

website/fuse.php
<?php

/* fuse.php: the main Fuse homepage
   Copyright (c) 1999-2010 Darren Salt, Philip Kendall

   $Id$

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 2 of the License, or
   (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License along
   with this program; if not, write to the Free Software Foundation, Inc.,
   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

   Author contact information:

   E-mail: philip-fuse@shadowmagic.org.uk

*/

include ".inc/fuse.inc";

fuse_title ("Fuse - the Free Unix Spectrum Emulator", true);
fuse_menu_heading ("Fuse");

#<!-- ======================================================= -->

$img = mk_image('screens/f-awm.png', 330, 292);
fuse_section (NULL, "What is it?", <<<END_SECTION
  $img
  <p>Fuse (the Free Unix Spectrum Emulator) was originally, and somewhat unsurprisingly, a ZX Spectrum emulator for Unix. However, it has now also been ported to Mac OS X, which may or may not count as a Unix variant depending on your advocacy position. It has also been ported to Windows, the Wii, AmigaOS and MorphOS, which are definitely not Unix variants.</p>

END_SECTION
);

#<!-- ======================================================= -->

$img = mk_image('screens/f-large.png', 650, 532, 4);
fuse_section ("Features", "What features does it have?", <<<END_SECTION
  $img
  <ul>
   <li>Accurate 16K, 48K (including the NTSC variant), 128K, +2, +2A and +3 emulation.</li>
   <li>Working +3e, SE, TC2048, TC2068, TS2068, Pentagon 128, Pentagon "512" (Pentagon 128 modified for extra memory), Pentagon 1024 and Scorpion ZS 256 emulation.</li>
   <li>Runs at true Speccy speed on any computer you're likely to try it on.</li>
   <li>Support for loading from .tzx files, including accelerated loading.</li>
   <li>Sound (on Windows and Mac OS X, and on systems supporting ALSA, the Open Sound System, SDL or OpenBSD/Solaris's <tt>/dev/audio</tt>).</li>
   <li>Kempston joystick emulation.</li>
   <li>Emulation of the various printers you could attach to the Spectrum.</li>
   <li>Support for the RZX input recording file format, including 'competition mode'.</li>
   <li>Emulation of the Currah &mu;Source, DivIDE, Fuller audio box, Interface 1, Kempston mouse, SpecDrum, Spectrum +3e, ZXATASP and ZXCF interfaces.</li>
   <li>Emulation of the Beta 128, +D, Didaktik 80/40, DISCiPLE and Opus Discovery interfaces.</li>
   <li>Emulation of the Spectranet and SpeccyBoot interfaces.</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section ("What's lacking?", "What is it lacking?", <<<END_SECTION
  <ul>
   <li>Quite a lot! However, it's a lot better than it used to be...</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

$img = mk_image('screens/f-rzx.png', 330, 266);
fuse_section ("Download", "Downloads", <<<END_SECTION
  $img
  <p>Fuse is licensed under the <a href="COPYING">GNU General Public License</a>, version 2 or later. Please read this before downloading Fuse if you're not already familiar with it.</p>

  <h3>Unix</h3>
  <p>Packages are available for some Unix distributions; in general, any problems which are specific to the packages should be sent to the package maintainer.</p>
  <ul>
   <li><a href="https://www.archlinux.org/">Arch Linux</a> users can get version 1.2.0 from AUR as a <a href="https://aur.archlinux.org/packages/fuse-emulator/">PKGBUILD</a>.</li>
   <li><a href="http://packages.debian.org/search?keywords=fuse-emulator">Official packages</a> of 1.2.1 for <a href="http://www.debian.org/">Debian</a>, maintained by Alberto Garcia, are available.</li>
   <li><a href="http://fedoraproject.org/">Fedora</a> has <a href="https://apps.fedoraproject.org/packages/fuse-emulator/">packages</a> of 1.2.1 available by Lucian Langa.</li>
   <li><a href="http://www.freebsd.org/">FreeBSD</a> has a port of 1.2.1 available as <a href="http://svnweb.freebsd.org/ports/head/emulators/fuse/">emulators/fuse</a>.</li>
   <li><a href="https://www.gentoo.org/">Gentoo</a> users have an <a href="https://packages.gentoo.org/packages/app-emulation/fuse">ebuild</a> of 1.1.1 available.</li>
   <li><a href="http://www.netbsd.org/">NetBSD</a> users can get version 1.1.1 from <a href="http://www.pkgsrc.org/">pkgsrc</a> (the NetBSD Packages Collection) as <a href="http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/emulators/fuse-emulator/">emulators/fuse-emulator</a> and <a href="http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/emulators/fuse-emulator-utils/">emulators/fuse-emulator-utils</a>.</li>
   <li><a href="http://www.openbsd.org/">OpenBSD</a> users have version 1.1.1 available as the fuse and fuse-utils package, with thanks to Anthony J. Bentley.</li>
   <li><a href="https://www.opensuse.org/">openSUSE</a> users have version 1.2.1 as the <a href="https://build.opensuse.org/package/show/Emulators/Fuse">Fuse</a> package from Emulators repository.</li>
   <li>The <a href="https://www.pld-linux.org/">Polish Linux Distribution</a> has packages of <a href="http://sophie.zarb.org/search?search=fuse-gtk&amp;type=fuzzyname&amp;deptype=P&amp;distribution=PLD&amp;release=th&amp;arch=">Fuse</a> 1.2.0 and the <a href="http://sophie.zarb.org/search?search=fuse-utils&amp;type=fuzzyname&amp;deptype=P&amp;distribution=PLD&amp;release=th&amp;arch=">utilities</a>.</li>
   <li><a href="http://www.slackware.com/">Slackware</a> users can get version 1.2.1 from SlackBuilds as a <a href="http://slackbuilds.org/result/?search=fuse-emulator">build script</a>.</li>
   <li><a href="http://www.ubuntu.com/">Ubuntu</a> has <a href="https://launchpad.net/ubuntu/+source/fuse-emulator">packages</a> of 1.2.1 available by Alberto Garcia.</li>
  </ul>
  <p>Packages of older versions of Fuse are also available for some other distributions:</p>
  <ul>
   <li>Nokia's <a href="http://maemo.org/">Maemo</a> platform has a port of <a href="http://maemo.org/downloads/product/OS2008/fuse-emulator/">1.0.0</a> available by Alberto Garcia.</li>
   <li>Mandriva packages of 0.10.0.2 were available from the <a href="http://plf.zarb.org/">Penguin Liberation Front</a>.</li>
  </ul>

  <h3>Mac OS X</h3>
  <p>A native port to OS X by <a href="mailto:fredm@spamcop.net">Fredrick Meunier</a> is available on its own SourceForge project <a href="http://fuse-for-macosx.sourceforge.net/">here</a>, as well as a Spotlight importer for Mac OS X 10.4 Tiger users. Alternatively, the original version of Fuse will compile on OS X 10.3 (Panther) or later.</p>

  <h3>Windows</h3>
  <p>A port to Windows of 1.2.2 by Sergio Baldovi is available <a href="https://sourceforge.net/projects/fuse-emulator/files/fuse/1.2.2/fuse-1.2.2-win32-setup.exe/download">here</a>, and the utilities are available <a href="https://sourceforge.net/projects/fuse-emulator/files/fuse-utils/1.2.0/fuse-utils-1.2.0-win32.zip/download">here</a>.</p>

  <h3>Android</h3>
  <p>BogDan Vatra has ported Fuse 1.2.1 to Android OS, which could run on smartphones, tablets and TVs. Sources are available from <a href="https://github.com/bog-dan-ro/spectacol/releases/tag/v2.5.0">GitHub</a> and binaries from <a href="https://play.google.com/store/apps/details?id=eu.licentia.games.spectacol">Google Play</a>.</p>

  <h3>Haiku</h3>
  <p>Adrien Destugues has ported Fuse 1.1.1 to Haiku, available from haikuports as <a href="https://github.com/haikuports/haikuports/tree/master/app-emulation/fuse">app-emulation/fuse</a> and <a href="https://github.com/haikuports/haikuports/tree/master/app-emulation/fuse-utils">app-emulation/fuse-utils</a>.</p>

  <h3>AmigaOS 4</h3>
  <p>Chris Young has ported Fuse 1.0.0.1 to AmigaOS 4, with binaries available from Aminet as <a href="http://aminet.net/package/misc/emu/fuse.lha">misc/emu/fuse.lha</a>.</p>

  <h3>MorphOS</h3>
  <p>Q-Master has ported Fuse 0.10.0.1 to MorphOS, with <a href="http://www.amirus.org.ru/files/fuse_0.10.0.1_morphos.lha">binaries available from AmiRUS</a>.</p>

  <h3>PSP</h3>
  <p>Akop Karapetyan has ported Fuse to the PSP. Binaries and source, based on Fuse 0.10.0.1, are available from the <a href="http://psp.akop.org/fuse">Fuse PSP</a> page.</p>

  <h3>Wii</h3>
  <p>A Wii port, based on work by Bj&ouml;rn Giesler, is available from <a href="http://wiibrew.org/wiki/Fuse">WiiBrew</a>. This is based on what is essentially 0.10.0.2.</p>

  <h3>Gizmondo</h3>
  <p>A port of 0.9.0 to the Gizmondo tablet is available. The source was available via <a href="https://web.archive.org/web/20090308111024/http://opensvn.csie.org/GizPorts/trunk/sdlport/fuzegiz/">csie.org</a>.</p>

  <h3>GP2X</h3>
  <p>Ben O'Steen has made a GP2X port, based on Fuse 0.6. Binaries and source are available from <a href="http://web.archive.org/web/20090905033548/http://www.zen71790.zen.co.uk/#fuse">his homepage</a>.</p>

  <h3>XBox</h3>
  <p>Crabfists's has made an Xbox port, based on Fuse 0.6. Binaries and source are available from the <a href="https://sourceforge.net/projects/fusex/">FuseX project</a> at SourceForge.</p>

  <h3>PocketPC</h3>
  <p>Anders Holmberg's ported Fuse 0.4 to the PocketPC as <a href="http://pocketclive.emuunlim.com/">PocketClive</a>.</p>

  <h3>Windows Mobile Smartphone</h3>
  <p>Keith Orbell's then ported PocketClive to the Smartphone as <a href="http://web.archive.org/web/20090418170517/http://www.aooa27.dsl.pipex.com/FuseSP.htm">FuseSP</a>.</p>

END_SECTION
);

#<!-- ======================================================= -->

$img = mk_image('screens/f-nongtk.png', 330, 266);
fuse_section ("Source", "Source", <<<END_SECTION
  <h3>Installing Fuse</h3>
  $img
  <ul>
   <li>First, check the requirements below and ensure all the libraries you want/need are installed.</li>
   <li>Secondly, install <a href="libspectrum.php">libspectrum</a>.</li>
   <li>Get the <a href="https://sourceforge.net/projects/fuse-emulator/files/fuse/1.2.2/fuse-1.2.2.tar.gz/download">source code</a> (<a href="https://sourceforge.net/projects/fuse-emulator/files/fuse/1.2.2/fuse-1.2.2.tar.gz.sig/download">PGP signature</a>).</li>
   <li>The utilities which were previously packaged with Fuse are now available in their <a href="https://sourceforge.net/projects/fuse-emulator/files/fuse-utils/1.2.2/fuse-utils-1.2.2.tar.gz/download">own package</a> (<a href="https://sourceforge.net/projects/fuse-emulator/files/fuse-utils/1.2.2/fuse-utils-1.2.2.tar.gz.sig/download">PGP signature</a>). Note that
you'll still need <a href="libspectrum.php">libspectrum</a> installed to run these.</li>
   <!--
   <li>The above are also mirrored at <a href="http://www.worldofspectrum.org/">World of Spectrum</a>: <a
href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/fuse-1.2.2.tar.gz">Fuse source</a> (<a
href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/fuse-1.2.2.tar.gz.asc">signature</a>), <a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/fuse-utils-1.2.2.tar.gz">utils source</a> (<a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/fuse-utils-1.2.2.tar.gz.asc">signature</a>).</li>
   -->
   <li>The source code releases above are signed with the <a href="http://wwwkeys.pgp.net:11371/pks/lookup?op=get&amp;search=0xD0767AB9">Fuse Release Key, ID <tt>D0767AB9</tt></a>, which has fingerprint <tt>064E 0BA9 688F 2699 3267 B1E5 1043 EEEB D076 7AB9</tt>. This is different from the key used to sign the 0.6.0(.1) releases as I forgot the passphrase for that key <tt>:-(</tt>.</li>
  </ul>

  <h3>Requirements</h3>
  <dl>
   <dt>Required:</dt>
   <dd>
    <ul>
     <li>X, <a href="http://www.libsdl.org/">SDL</a>, <a href="http://www.svgalib.org/">svgalib</a> or framebuffer support. If you have <a href="http://www.gtk.org/">GTK+</a> installed, you'll get a (much) nicer user interface under X.</li>
     <li><a href="libspectrum.php">libspectrum</a>: the Spectrum emulator file format and information library.</li>
    </ul>
   </dd>
   <dt>Optional:</dt>
   <dd>
    <ul>
     <li><a href="http://www.gnu.org/software/libgcrypt/">libgcrypt</a>: the ability to digitally sign RZX files (note that Fuse requires version 1.1.42 or later).</li>
     <li><a href="http://www.libpng.org/pub/png/libpng.html">libpng</a>: the ability to save screenshots.</li>
     <li><a href="http://xmlsoft.org/">libxml2</a>: the ability to load and save Fuse's current configuration and capture BASIC video functions to SVG.</li>
     <li><a href="http://www.libsdl.org/">SDL</a> or <a href="http://freecode.com/projects/libjsw">libjsw</a>: allow joystick input to be used (not required for joystick emulation).</li>
     <li><a href="http://www.zlib.net/">zlib</a>: support for compressed RZX files.</li>
     <li><a href="http://www.bzip.org/">libbzip2</a>: support for certain compressed files.</li>
     <li><a href="http://www.68k.org/~michael/audiofile/">libaudiofile</a>: support for loading from .wav files.</li>
     <li>Versions previous to 0.10.0 used John Elliott's <a href="http://www.seasip.demon.co.uk/Unix/LibDsk/">lib765 and libdsk</a> for the +3 support. 0.10.0 and newer include this support natively, so these libraries are no longer necessary (or used).</li>
    </ul>
   </dd>
  </dl>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section (NULL, "What's new?", <<<END_SECTION
  <h3>1.2.2</h3>
  <ul>
    <li>Support for loading the first tape, snapshot, dock cartridge or RZX file found inside .zip files.</li>
    <li>Support for auto-booting TR-DOS disk images without a boot file.</li>
    <li>Fix +3 disk autoload.</li>
    <li>Various minor bugfixes.</li>
  </ul>

  <h3>1.2.1</h3>
  <ul>
    <li>Add Z80 registers, last byte written to the ULA, tstates since interrupt and the primary and secondary memory control ports as debugger variables.</li>
    <li>Extend breakpoints on paging events to more peripherals: Beta 128, +D, Didaktik 80, DISCiPLE, Opus Discovery and SpeccyBoot.</li>
    <li>Fix crash on widget UIs when hitting the close icon on the title bar several times.</li>
    <li>Fix loading bugs when the detect loaders feature is being used.</li>
  </ul>

  <h3>1.2.0</h3>
  <ul>
    <li>Emulation of the Currah &mu;Source and Didaktik 80/40 interfaces.</li>
    <li>Capture BASIC video functions to SVG file.</li>
    <li>Allow continuing RZX recordings.</li>
    <li>Support bash completion.</li>
    <li>Really lots of bugfixes and miscellaneous improvements.</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section (NULL, "Development", <<<END_SECTION
  <p>If you're just want news of new versions and the like, the (low volume) <a href="https://lists.sourceforge.net/lists/listinfo/fuse-emulator-announce">fuse-emulator-announce</a> list is available. If you're interested in the development of Fuse, this is coordinated via the <a href="https://lists.sourceforge.net/lists/listinfo/fuse-emulator-devel">fuse-emulator-devel</a> list and the <a href="https://sourceforge.net/projects/fuse-emulator/">project page</a> on SourceForge.</p>

  <p>The latest version of Fuse is always available by checking out the 'trunk/fuse' directory from the <a href="https://sourceforge.net/p/fuse-emulator/code/">Subversion repository</a> on SourceForge. Note that this isn't guaranteed to compile, let alone work properly. Also, don't expect any support for this version! (You'll also need libspectrum from Subversion; this is in the 'trunk/libspectrum' directory). Similarly, the utilities are available in the 'trunk/fuse-utils' directory.</p>

  <p>One thing which isn't in the SourceForge tracking system (and is now very outdated):</p>

  <ul>
   <li>David Gardner has produced a patch to give <a href="fuse-xvideo.patch.bz2">XVideo support</a> for the Xlib UI, allowing arbitrary sized windows.</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section ("Related projects", "Are there any related projects?", <<<END_SECTION
  <ul>
   <li><a href="libspectrum.php">libspectrum</a> is the library used by Fuse to handle various file formats.</li>
   <li>Mike Wynne's ZX81 emulator, <a href="http://www.chuntey.com/eightyone/">EightyOne</a> is also using Fuse's Z80 core.</li>
   <li>Matthew Westcott's <a href="http://matt.west.co.tt/spectrum/jsspeccy/">JSSpeccy</a> uses a Z80 core based on translating Fuse's core to Javascript.</li>
   <li>Alexander Shabarshin's <a href="http://web.archive.org/web/20030316123953/http://robots.ural.net/nedopc/sprinter/">SPRINT</a>, an emulator of the <a href="http://www.interface1.net/zx/clones/peters.html">Peters Plus</a> super-Speccy, the <a href="http://web.archive.org/web/20040402004216/http://www.petersplus.com/sprinter/">Sprinter</a>. SPRINT is using Fuse's Z80 core for its CPU emulation.</li>
   <li><a href="https://sourceforge.net/projects/z80ex/">z80ex</a>, a Z80 emulation library based on Fuse's Z80 core, used by zemu and <a href="http://pocketspeccy.narod.ru/">PocketSpeccy</a>.</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_related ("libspectrum", "libspectrum.php");

fuse_footer ();
?>
